packages/controller-utils/src/caip.test.ts

```diff
@@ -0,0 +1,98 @@
+    expect(isEthCaipChainId('eip155:1337')).toBe(true);
+    expect(isEthCaipChainId('bip122:1')).toBe(false);
+    expect(isEthCaipChainId('eip1555:1')).toBe(false);
+    expect(isEthCaipChainId('eip:1')).toBe(false);
+    expect(isEthCaipChainId('eip:')).toBe(false);
+    expect(isEthCaipChainId('eip155')).toBe(false);
+  });
+
+  describe('buildEthCaipChainId', () => {
+    describe('valid hex chain id', () => {
+      it('returns a CAIP chain ID string for eip155', () => {
+        expect(buildEthCaipChainId('0x1')).toBe('eip155:1');
+        expect(buildEthCaipChainId('0x539')).toBe('eip155:1337');
+      });
+    });
+
+    describe('invalid hex chain id', () => {
+      it('throws an error', () => {
+        expect(() => buildEthCaipChainId('0xZZZ')).toThrow(
+          'Invalid chain ID "0xZZZ"',
+        );
+      });
+    });
+
+    describe('decimal chain id', () => {
+      it('returns a CAIP chain ID string for eip155', () => {
+        expect(buildEthCaipChainId('1')).toBe('eip155:1');
+        expect(buildEthCaipChainId('1337')).toBe('eip155:1337');
+      });
+    });
+
+    describe('invalid decimal chain id', () => {
+      it('throws an error', () => {
+        expect(() => buildEthCaipChainId('ZZZ')).toThrow(
+          'Invalid chain ID "ZZZ"',
+        );
+      });
+    });
+  });
+
+  describe('parseEthCaipChainId', () => {
+    it('returns the reference as is', () => {
+      expect(parseEthCaipChainId('eip155:1')).toBe('1');
+      expect(parseEthCaipChainId('eip155:1337')).toBe('1337');
+      expect(parseEthCaipChainId('eip155:abc')).toBe('abc');
+    });
+  });
+
+  describe('parseEthCaipChainIdHex', () => {
+    describe('valid CAIP chain ID', () => {
+      it('returns the reference parsed into hex string', () => {
+        expect(parseEthCaipChainIdHex('eip155:1')).toBe('0x1');
+        expect(parseEthCaipChainIdHex('eip155:1337')).toBe('0x539');
+      });
+    });
+
+    describe('invalid CAIP chain ID', () => {
+      it('throws an error', () => {
+        expect(() => parseEthCaipChainIdHex(':123')).toThrow(
+          'Invalid CAIP chain ID.',
+        );
+        expect(() => parseEthCaipChainIdHex('eip155:ABC')).toThrow(
+          'Invalid character',
+        ); // this comes from toHex(). What should we throw instead?
+      });
+    });
+  });
+
+  describe('parseEthCaipChainIdInt', () => {
+    describe('valid CAIP chain ID', () => {
+      it('returns the reference parsed into a number', () => {
+        expect(parseEthCaipChainIdInt('eip155:1')).toBe(1);
+        expect(parseEthCaipChainIdInt('eip155:1337')).toBe(1337);
+      });
+    });
+
+    describe('invalid CAIP chain ID', () => {
+      it('returns NaN', () => {
+        const invalidChainId = parseEthCaipChainIdInt(':123');
+        expect(Number.isNaN(invalidChainId)).toBe(true);
+        const invalidChainIdReference = parseEthCaipChainIdInt('eip155:ABC');
+        expect(Number.isNaN(invalidChainIdReference)).toBe(true);
```

Comment:
Should we throw here instead?

Reply:
hard to say. Currently in extension, the pattern i see is that we check for NaN in if guard statements after attempting to parseInt. Throwing would change this pattern from

```
const chainIdInt = parseInt(chainId, 10)

if (Number.isNaN(chainIdInt)) {
  // exit if NaN
  return
}

doSomething(chainIdInt)
```

to

```
let chainIdInt

try {
chainIdInt = parseEthCaipChainIdInt(caipChainId)
} catch (err) {
    // exit if NaN
}

doSomething(chainIdInt)
```

it seems like we're deciding if we should treat NaN as an exception or not